hazelcast/src/hazelcast/client/config.cpp

```diff
@@ -852,11 +851,20 @@ namespace hazelcast {
 
         const config::reliable_topic_config &client_config::get_reliable_topic_config(const std::string &name) {
             auto it = reliable_topic_config_map_.find(name);
-            if (reliable_topic_config_map_.end() == it) {
-                reliable_topic_config_map_[name] = config::reliable_topic_config(name.c_str());
+            if (it != reliable_topic_config_map_.end()) {
+                return it->second;
             }
 
-            return reliable_topic_config_map_[name];
+            return reliable_topic_config_map_.emplace(name, config::reliable_topic_config(name)).first->second;
```

Comment:
```suggestion
            return reliable_topic_config_map_.emplace(name, name).first->second;
```
emplace will construct the config object in-place.